extensions/usage-metrics-dashboard/app.R

```diff
@@ -274,25 +275,82 @@ ui <- function(request) {
 server <- function(input, output, session) {
   # Set up Connect client; handle error if Visitor API Key integration isn't
   # present.
+
+  publisher_client <- connect()
+
+  selected_integration_guid <- reactiveVal(NULL)
+  observeEvent(input$auto_add_integration, {
+    print("Clicked auto add integration button")
+    auto_add_integration(publisher_client, selected_integration_guid())
+    # Hard refresh so that the sidebar gets the up to date info
+    runjs("window.top.location.reload(true);")
+  })
+
   client <- NULL
   tryCatch(
     client <- connect(
       token = session$request$HTTP_POSIT_CONNECT_USER_SESSION_TOKEN
     ),
     error = function(e) {
-      showModal(modalDialog(
-        title = "Additional Setup Required",
-        footer = NULL,
-        HTML(paste(
-          "In the Access panel to the right, click <strong>\"Add integration\"</strong>,",
-          "then select a <strong>Visitor API Key</strong> integration.",
-          "If you don't see one in the list, an administrator must enable this feature on your Connect server.",
-          "See the <a href='https://docs.posit.co/connect/admin/integrations/oauth-integrations/connect/' target='_blank'>Admin Guide</a> for setup instructions.",
+      eligible_integrations <- get_eligible_integrations(publisher_client)
+      selected_integration <- slice_head(eligible_integrations, n = 1)
+      selected_integration_guid(selected_integration$guid)
+
+      message <- paste0(
+        "This content requires a <strong>Visitor API Key</strong> ",
+        "integration to show users the content they have access to."
```

Comment:
```suggestion
        "This content requires a Publisher or Admin <strong>Visitor API Key</strong> ",
        "integration to show users the content they have access to."
```

Minor suggestion to add the "Publisher" / "Admin" prior to the bolded text in the hopes it gets read.

Reply:
After today's discussion, do you think we need to mention this detail to users? This text shows all the time, and I think that if an appropriate integration exists, we don't need to show those things to the user.

I do hear what you're saying though, good to have these two words more noticeable. Other options:

- We could have the entire message be conditional, and this detail could go only in the message we show if no integration exists. I think that could be better, I'm having second thoughts about the whole wording today.
- We could bold the words where they appear on line 324, the instructions on how to create a new integration.

Reply:
With the fully automated add flow I do think the distinction is a lot less important. On the no-integration-exists case I do think we need to be clear about what is needed.

Reply:
I know we discussed a fully automated flow in the demo session, but I thought afterwards we landed on still liking to have an interstitial with a button press. Happy to try fully automating it tomorrow though.